include/swift/AST/DiagnosticsSema.def

```diff
@@ -784,9 +784,10 @@ ERROR(reserved_member_name,none,
       "type member must not be named %0, since it would conflict with the"
       " 'foo.%1' expression", (DeclName, StringRef))
 
-ERROR(invalid_redecl_by_optionality,none,
-      "invalid redeclaration of %0 which differs only by the kind of optional passed as an inout argument (%1 vs. %2)",
-      (DeclName, Type, Type))
+NOTE(invalid_redecl_by_optionality_note,none,
+      "parameter with %select{implicitly unwrapped |}0optional type is not different "
```

Comment:
Nit: Can we massage this message so that IUOs aren't referred to as a type? The underlying principle here as exposed to users is that IUOs _aren't_ a distinct type; it's not merely that we consider the parameter to be the same ("not different"—I wonder if that wording can be simplified also), but also the parameter _type_.

Reply:
@xwu So something like: _parameter with implicitly unwrapped optional is not different from original optional parameter_. I am not sure how I would have "not different" differently though, any suggestions?

Reply:
"Not different" = "same"

Reply:
Ah yeah, that totally makes sense, I thought this message would totally need to be replaced by something even more simpler. Thanks for the suggestion, so the new message then would be: _parameter with implicitly unwrapped optional is same as original optional parameter_ @xwu

Reply:
How about "implicitly unwrapped optional parameter is of same type as optional parameter"

Reply:
Changed it